webapp/publisher/snaps/build_views.py

```diff
@@ -257,3 +270,33 @@ def post_trigger_build(snap_name):
     return flask.redirect(
         flask.url_for(".get_snap_builds", snap_name=snap_name)
     )
+
+
+@csrf.exempt
+def post_github_webhook(snap_name):
+    github_repo = flask.request.json["repository"]["url"]
+    lp_snap = launchpad.get_snap_by_store_name(snap_name)
+
+    # Check that this is the repo for this snap
+    if lp_snap["git_repository_url"] != github_repo:
+        return ("The repository does not match the one used by this Snap", 403)
+
+    github = GitHubAPI()
+    gh_owner, gh_repo = github_repo[19:].split("/")
```

Comment:
If you use `github_repo = flask.request.json["repository"]["full_name"]` you don't need to slice.

Alternatively `gh_owner = flask.request.json["repository"]["owner"]["logn"]` and  `gh_repo = flask.request.json["repository"]["name"]` should give you what you want without worrying about slicing or splitting.

Reply:
Hey @jpmartinspt , I need the full URL, I will keep using the URL property. For the owner and the repo name, you are right. I updated the code.